test/qtismtest/data/AssessmentTestTest.php

```diff
@@ -114,4 +115,168 @@ public function testIsExclusivelyLinear()
+        $possible_paths[] = new AssessmentItemRefCollection([$itemq1, $itemq2, $itemq3, $itemq6]);
+        $possible_paths[] = new AssessmentItemRefCollection([$itemq1, $itemq3, $itemq6]);
+
+        foreach ($possible_paths as $path) {
+            $possible_paths2[] = $path->getArrayCopy();
+        }
+
+        $this->assertEquals($possible_paths, $test->getPossiblePaths(false));
+        $this->assertEquals($possible_paths2, $test->getPossiblePaths(true));
+    }
+
+    public function testPossiblePathswithPreCondition()
+    {
+        // Case 1
+
+        $doc = new XmlDocument();
+        $doc->load(self::samplesDir() . 'custom/tests/branchingpathwithpre.xml');
+        $test = $doc->getDocumentComponent();
+
+        $itemq1 = $doc->getDocumentComponent()->getComponentByIdentifier('Q01');
+        $itemq2 = $doc->getDocumentComponent()->getComponentByIdentifier('Q02');
+        $itemq3 = $doc->getDocumentComponent()->getComponentByIdentifier('Q03');
+        $itemq4 = $doc->getDocumentComponent()->getComponentByIdentifier('Q04');
+        $itemq5 = $doc->getDocumentComponent()->getComponentByIdentifier('Q05');
+
+        $possible_paths = array();
+        $possible_paths[] = new AssessmentItemRefCollection([$itemq1, $itemq2, $itemq3, $itemq4, $itemq5]);
+        $possible_paths[] = new AssessmentItemRefCollection([$itemq1, $itemq3, $itemq4, $itemq5]);
+        $possible_paths[] = new AssessmentItemRefCollection([$itemq1, $itemq2, $itemq3, $itemq5]);
+        $possible_paths[] = new AssessmentItemRefCollection([$itemq1, $itemq3, $itemq5]);
+        $possible_paths[] = new AssessmentItemRefCollection([$itemq1, $itemq2, $itemq3, $itemq4]);
+        $possible_paths[] = new AssessmentItemRefCollection([$itemq1, $itemq3, $itemq4]);
+        $possible_paths[] = new AssessmentItemRefCollection([$itemq1, $itemq2, $itemq3]);
+        $possible_paths[] = new AssessmentItemRefCollection([$itemq1, $itemq3]);
+
+        $this->assertEquals($possible_paths, $test->getPossiblePaths(false));
+
+        // Case with duplicates
+
+        $doc = new XmlDocument();
+        $doc->load(self::samplesDir() . 'custom/tests/testnoduplicatepaths.xml');
+        $test = $doc->getDocumentComponent();
+
+        $itemq1 = $doc->getDocumentComponent()->getComponentByIdentifier('Q01');
+        $itemq2 = $doc->getDocumentComponent()->getComponentByIdentifier('Q02');
+        $itemq3 = $doc->getDocumentComponent()->getComponentByIdentifier('Q03');
+
+        $possible_paths = array();
+        $possible_paths[] = new AssessmentItemRefCollection([$itemq1, $itemq2, $itemq3]);
+        $possible_paths[] = new AssessmentItemRefCollection([$itemq1, $itemq3]);
+
+        $this->assertEquals($possible_paths, $test->getPossiblePaths(false));
+    }
+
+    // Testing special cases
+
+    public function testRecursiveBranching()
+    {
+        $this->setExpectedException(
```

Comment:
I know this is how we do usually across the QTI-SDK. Unfortunately, PHPUnit 6.X.X removed this method (previously deprecated bad luck!).

Could you please investigate what could be the most effective "new" way to achieve the same assertion, but compliant with PHPUnit 5, 6, ... Thanks !

(this will be helpful for me as well to replicate on the other parts of the software, and there are many :) !